src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -697,13 +705,26 @@ def _infer_sharing_attributes(self):
             raise GenerationError("OMPParallelClause cannot correctly generate"
                                   " the private clause when its default "
                                   "data sharing attribute in its default "
-                                  "clause is not shared.")
+                                  "clause is not 'shared'.")
+
+        # TODO #598: Improve the handling of scalar variables, there are
+        # remaining issues when we have accesses after the parallel region
+        # of variables that we currently declare as private. This should be
+        # lastprivate.
+        # e.g:
+        # !$omp parallel do <- will set private(ji, my_index)
+        # do ji = 1, jpk
+        #   my_index = ji+1
+        #   array(my_index) = 2
+        # enddo
+        # #end do
+        # call func(my_index) <- my_index does not have a value
```

Comment:
Strictly speaking, won't it have the value given it by the last thread to write to it?

Reply:
Not if it is private, at least the standard doesn't say so. I have updated to "my_index has not been updated"